pkg/products/threescale/reconciler.go

```diff
@@ -2911,6 +2924,30 @@ func (r *Reconciler) ensureDeploymentConfigsReady(ctx context.Context, serverCli
 	return integreatlyv1alpha1.PhaseCompleted, nil
 }
 
+func (r *Reconciler) addResourceRequests(ctx context.Context, client k8sclient.Client) (integreatlyv1alpha1.StatusPhase, error) {
+	for _, name := range deploymentConfigsThatNeedRequestAnnotation {
+		deploymentConfig := &appsv1.DeploymentConfig{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: r.Config.GetNamespace(),
+			},
+		}
+		_, err := controllerutil.CreateOrUpdate(ctx, client, deploymentConfig, func() error {
+			if deploymentConfig.Spec.Template.ObjectMeta.Annotations == nil {
+				deploymentConfig.Spec.Template.ObjectMeta.Annotations = map[string]string{}
+			}
+			deploymentConfig.Spec.Template.ObjectMeta.Annotations["marin3r.3scale.net/resources.requests.cpu"] = "190m"
+			deploymentConfig.Spec.Template.ObjectMeta.Annotations["marin3r.3scale.net/resources.requests.memory"] = "90Mi"
+			return nil
+		})
+		if err != nil {
+			return integreatlyv1alpha1.PhaseFailed, fmt.Errorf("failed to add resource annotation to DeploymentConfig %s: %v", deploymentConfig.Name, err)
+		}
+	}
+
+	return integreatlyv1alpha1.PhaseCompleted, nil
+}
+
```

Comment:
Would adding these annotations cause a subsequent pod rollout ?

If so, I'm wondering would it be better to add the annotations into https://github.com/integr8ly/integreatly-operator/blob/master/pkg/resources/ratelimit/envoy.go#L100-L104 instead ? Downside is readability though 🤔 

I think if we do these seperately, and if they cause muliple pod rollouts, the installation would be generally longer.

Alternatively maybe move the function up closer to after`r.reconcileRatelimitingTo3scaleComponents` instead of near the end of the reconcile loop 🤔

Reply:
I didn't notice the installation taking longer or see multiple rollouts but I also wasn't looking for them. I think adding the annotations to the [patchDeploymentConfig](https://github.com/integr8ly/integreatly-operator/blob/master/pkg/resources/ratelimit/envoy.go#L100-L104) function like you suggested makes more sense. I'll get started on testing the change.